cirq/sim/sparse_simulator_test.py

```diff
@@ -605,6 +605,28 @@ def _apply_unitary_(self, args: cirq.ApplyUnitaryArgs):
     assert not initial_state is result.state_vector()
 
 
+def test_does_not_modify_initial_state():
+    q0 = cirq.LineQubit(0)
+    simulator = cirq.Simulator()
+
+    class InPlaceUnitary(cirq.SingleQubitGate):
+
+        def _has_unitary_(self):
+            return True
+
+        def _apply_unitary_(self, args: cirq.ApplyUnitaryArgs):
+            args.target_tensor[0], args.target_tensor[1] = (
+                args.target_tensor[1], args.target_tensor[0])
+            return args.target_tensor
+
+    circuit = cirq.Circuit(InPlaceUnitary()(q0))
+
+    initial_state = np.array([1, 0], dtype=np.complex64)
+    _ = simulator.simulate(circuit, initial_state=initial_state)
+    np.testing.assert_array_almost_equal(np.array([1, 0], dtype=np.complex64),
+                                         initial_state)
```

Comment:
Optional nit: I'd add a second assert on the result (ignored on l.625) to ensure simulate had any effect at all (if it didn't have an effect then even when simulate was making a copy the current assert might have succeed).

Reply:
done